module3/projects/rails_engine/requirements.md

```diff
@@ -135,115 +210,177 @@ Example JSON response for the Merchant resource:
 }
 ```
 
-### Show Record
+## RESTful: Fetch a single record
 
-This endpoint renders a JSON representation of the corresponding record.
+This endpoint for items and merchants should:
 
-The URI should follow this pattern: `GET /api/v1/<resource>/:id`
+* render a JSON representation of the corresponding record, if found
+* follow this pattern: `GET /api/v1/<resource>/:id`
 
-Example JSON response for the Merchant resource:
+Example JSON response for the Item resource:
 
 ```json
 {
   "data": {
     "id": "1",
-    "type": "merchant",
+    "type": "item",
     "attributes": {
-      "name": "Store Name"
+      "name": "Super Widget",
+      "description": "A most excellent widget of the finest crafting"
+      "unit_price": 109.99
     }
   }
 }
 ```
 
-### Create Record
+Note that the `unit_price` is sent as numeric data, and not string data.
 
-This endpoint should create a record and render a JSON representation of the new record.
 
-The URI should follow this pattern: `POST /api/v1/<resource>`
+## RESTful: Create an Item
 
-The body should follow this pattern:
+This endpoint should:
+
+* create a record and render a JSON representation of the new Item record.
+* follow this pattern: `POST /api/v1/items`
+* accept the following JSON body with only the following fields:
 
 ```
 {
-  "attribute1": "value1",
-  "attribute2": "value2",
+  "name": "value1",
+  "description": "value2",
+  "unit_price": 100.99
 }
 ```
+(Note that the unit price is to be sent as a numeric value, not a string.)
 
-Example JSON response for the Merchant resource:
+* return an error if any attribute is missing
+* should ignore any attributes sent by the user which are not allowed
+
+Example JSON response for the Item resource:
 
 ```json
 {
   "data": {
-    "id": "1",
-    "type": "merchant",
+    "id": "16",
+    "type": "item",
     "attributes": {
-      "name": "Store Name"
+      "name": "Widget",
+      "description": "High quality widget",
+      "unit_price": 100.99,
+      "created_at": "2020-12-18T13:42:56 UTC",
+      "updated_at": "2020-12-18T13:42:56 UTC"
     }
   }
 }
 ```
```

Comment:
Could we ask them to return the same data as fetching a single item, leaving off the created_at and updated_at so they can reuse the same serializer? Also, should we ask them to specify the status code of 201 for the response?

Reply:
ah yes, I meant to remove the timestamps from these